### PR TITLE
Hours editor code sharing

### LINKED ISSUE #5927: Reduce duplication of shared code in hours editor
> Could we move allDays and toggleDay out of the iOS and android functional calls? I think they're identical for each platform.

_Originally posted by @drewvolz in https://github.com/StoDevX/AAO-React-Native/pull/5926#discussion_r851837107_

### PR DESCRIPTION
**Pull request by drewvolz**
Closes #5927 –– I recommend [viewing without whitespace.](https://github.com/StoDevX/AAO-React-Native/pull/5928/files?diff=split&w=1)

- Fixup prop types for hours editor
- Reuse toggleDay and allDays in hours editor
- Combine render functions and return platform specific view

### CODE REVIEW

File: source/views/building-hours/report/editor.tsx [hunk 1]
```diff
@@ -106,7 +105,7 @@ function WeekTogglesIOS(props: WeekTogglesProps) {
 
 	let allDays: DayOfWeekEnumType[] = ['Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa', 'Su']
 
-	return (
+	let WeekTogglesIOS = (
```
Comment (hawkrives) [inline source/views/building-hours/report/editor.tsx]:
```suggestion
   if (Platform.OS === 'ios') {
      return (…)
   } else if (Platform.OS === 'android') {
      return (…)
   } else {
      return null 
   } 
```

Thoughts?

Comment (drewvolz) [reply source/views/building-hours/report/editor.tsx]: Nice, let's go with that.

Comment (drewvolz) [reply source/views/building-hours/report/editor.tsx]: Is this a good case for Platform.select?

```js
return Platform.select({
	ios: (
		<Row style={styles.weekToggles}>
		(…)
		</Row>
	),
	android: (
		<View>
		(…)
		</View>
	),
})
```

Comment (hawkrives) [reply source/views/building-hours/report/editor.tsx]: Sure!

**Review by rye — APPROVED** [review]

**Review by hawkrives — APPROVED** [review]

**Pull request merged**